src/oci/registry.rs

```diff
@@ -0,0 +1,24 @@
+use crate::oci::image::manifest::Manifest;
+use std::error;
+
+pub fn get_manifest(path: String) -> Result<Manifest, Box<dyn error::Error>> {
+    let res = ureq::get("http://httpbin.org/get").call().into_string()?;
+    let m: Result<Manifest, serde_json::Error> = serde_json::from_str(&res);
+
+    match m {
+        Ok(m) => Ok(m),
+        Err(e) => Err(Box::new(e)),
```

Comment:
I believe you can get rid of the match by calling map_err on the match. 
Do you have to box the error?

Reply:
http://www.sheshbabu.com/posts/rust-error-handling/

Reply:
I was not happy with this and found something even better.
Give it a look ! ;)